Automattic/newspack-image-credits
# feat: add settings page, and show placeholder if no credit

Adds a new settings page for Image Credits options under the **WP Admin > Settings > Media** menu (placed there because this plugin has no dependencies on other Newspack products). New options available:

* **Image Credit Class Name:** Lets you set the CSS class name given to the `span` that wraps image credit strings. This defaults to `image-credit`, which was previously hard-coded, so it shouldn't change behavior for sites that are already using the plugin.
* **Image Credit Label:** Lets you change the label printed before image credits. This defaults to `Credit: `, which was previously hard-coded, so it shouldn't change behavior for sites that are already using the plugin. 
* **Placeholder Image:** Lets you choose an image from the media library to be displayed in place of any image that lacks a credit. This should help publishers avoid legal trouble in the event of messy migrations that fail to bring over all credit data. Will work for featured images, core/image blocks, and jetpack/slideshow blocks.

<img width="1433" alt="Screen Shot 2021-08-23 at 2 05 44 PM" src="https://user-images.githubusercontent.com/2230142/130513969-2ac0f87f-d622-4942-a4f3-a0d40d991f6a.png">

_Note: The options have been moved to the Media settings page instead of living in their own standalone page._

To test:

1. The first two options (class name and label) are straightforward: test with the default options, then with updated options, and then with empty options. Verify that the settings are respected on the front-end for all three contexts (featured image, core image block, Jetpack slideshow block).
2. For the placeholder image, first verify that without choosing any image, behavior is unchanged. Images are rendered as usual with or without a credit.
3. Choose a placeholder image from the Media Library, then save.
4. Now verify that any images that lack a credit are replaced by the chosen placeholder image (just the image, not the caption or anything else having to do with the image) in all contexts.
5. Add a credit to images that are missing them and verify that after a refresh, the images render instead of the placeholder.
6. Remove the placeholder in settings, save, and verify that all images behave as usual.
